pytorch_lightning/trainer/trainer.py

```diff
@@ -53,7 +53,7 @@ class Trainer(TrainerIOMixin,
     def __init__(self,
                  logger=True,
                  checkpoint_callback=True,
-                 early_stop_callback=True,
+                 early_stop_callback=False,
```

Comment:
why changing the default config?

Reply:
As I have written in the description we should discuss whether early stopping should be turned on or off by the default. I think that it is better to be turned off.

Reply:
Early stopping should default to True - this is the most common use case.

Reply:
The problem is that it is conditioned on the `val_loss` and it will work only if the user guess right the name. For this reason it seems to me that the better option is when the user deliberately enables early stopping. It is also very easy (just set `True`).

But if you insist I can suggest the following: we enable early stopping by default, but if there is no `val_loss` we will just warn the user that early stopping will not work and training will proceed with disabled early stopping. I just don't like that there will be some warnings when you run the trainer with the default settings :)